composer/trainer/trainer.py

```diff
@@ -3470,13 +3477,19 @@ def _eval_loop(
                                         )[0]
 
                             self.engine.run_event(Event.EVAL_BEFORE_FORWARD)
-
                             with _get_precision_context(
                                 self.state.precision,
                                 self.state.precision_config,
                                 self.state.deepspeed_enabled,
                             ):
-                                self.state.outputs = self._original_model.eval_forward(self.state.batch)
+                                # Disabling FP8 in eval mode and defaulting to BF16.
+                                # This is because FP8 in TE requires all eval data sizes to be divisible by 16 which does not hold for all evaluation datasets.
+                                # See https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/examples/fp8_primer.html for more info.
+                                with get_fp8_precision_context(
+                                    fp8_autocast_enabled=False,
+                                    precision_config=None,
+                                ):
```

Comment:
does this run in fp32 or b16?

Reply:
bf16. see run here: `fp8-llama3-8b-metamath-4ep-Ok7FKv`